src/WebJobs.Extensions.Twilio/Config/TwilioSMSConfiguration.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Twilio
+{
+    /// <summary>
+    /// Defines the configuration options for the Twilio binding.
+    /// </summary>
+    public class TwilioSmsConfiguration
+    {
+        internal const string AzureWebJobsTwilioAccountSidKeyName = "AzureWebJobsTwilioAccountSid";
+        internal const string AzureWebJobsTwilioAccountAuthTokenName = "AzureWebJobsTwilioAuthToken";
+
+        public TwilioSmsConfiguration()
+        {
+            var nameResolver = new DefaultNameResolver();
+            AccountSid = nameResolver.Resolve(AzureWebJobsTwilioAccountSidKeyName);
```

Comment:
Don't create your own NameResolver -- that prevents anyone from overriding it in the JobHostConfig. 

Building on my comment above -- if this class implements IExtensionConfigProvider, you can get the NameResolver in the Initialize() method and set things up. Again, see: [https://github.com/Azure/azure-webjobs-sdk-extensions/blob/master/src/WebJobs.Extensions.DocumentDB/Config/DocumentDBConfiguration.cs](https://github.com/Azure/azure-webjobs-sdk-extensions/blob/master/src/WebJobs.Extensions.DocumentDB/Config/DocumentDBConfiguration.cs)